src/main/java/com/abramchik/taskTwoCollections/hashMap/MyHashMapImpl.java

```diff
@@ -0,0 +1,228 @@
+package com.abramchik.taskTwoCollections.hashMap;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class MyHashMapImpl<K, V> implements MyHashMap<K, V> {
+
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] hashTable;
+    private int size = 0;
+    private float threshold;
+
+
+    public MyHashMapImpl() {
+        hashTable = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
+
+    private int hash(K key) {
+        int hash = 31;
+        hash = hash * 17 * key.hashCode();
+        return Math.abs(hash % hashTable.length);
+    }
+
+
+    @Override
+    public boolean put(K key, V value) {
+        if (size + 1 >= threshold) {
+            threshold *= 2;
+            arrayGrow();
+        }
+
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = hash(key);
+
+        if (hashTable[index] == null) {
+            return simpleAdd(index, newNode);
+        }
+
+        List<Node<K, V>> nodeList = hashTable[index].getNodes();
+
+        if (keyExists(key)) {
+            for (Node<K, V> node : nodeList) {
+                if (key.equals(node.getKey())) {
+                    node.setValue(value);
+                }
+            }
+        } else {
+            nodeList.add(newNode);
+            size++;
+        }
+        return true;
+    }
+
+    private void arrayGrow() {
+        Node<K, V>[] oldHashTable = hashTable;
+        hashTable = new Node[oldHashTable.length * 2];
```

Comment:
the same, it could be HASH_TABLE_MULTIPLICATOR or something like that

Reply:
done